clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/ops/UpsertAzureAppGatewayAtomicOperation.groovy

```diff
@@ -98,49 +99,75 @@ class UpsertAzureAppGatewayAtomicOperation implements AtomicOperation<Map> {
         // We are attempting to create a new application gateway
         description.credentials.resourceManagerClient.initializeResourceGroupAndVNet(description.credentials, resourceGroupName, virtualNetworkName, description.region)
```

Comment:
If customers are "bringing there own" virtual network, then we need to also make sure that initializeResourceGroupAndVNet doesn't create a new virtual network resource.